microcloud/test/includes/microcloud.sh

```diff
@@ -663,6 +729,84 @@ reset_systems() {
     else
       reset_system "${name}" "${num_disks}" "${num_ifaces}"
     fi
+
+    if [ -n "${public_ceph_subnet}" ]; then
+      if [ "${num_ifaces}" -lt 3 ]; then
+        echo "Impossible to set a dedicated Ceph public network for less than 3 network interfaces."
+        false
+      fi
+
+      ip=$(echo "${public_ceph_subnet}" | cut -d '/' -f 1)
+      range=$(echo "${public_ceph_subnet}" | cut -d '/' -f 2)
+      IFS='.' read i1 i2 i3 i4 <<EOF
+$ip
+EOF
+
+      ip_num=$(( (i1 * 256 * 256 * 256) + (i2 * 256 * 256) + (i3 * 256) + i4 ))
+      host_bits=$(( 32 - range ))
+      pow=1
+      for _ in $(seq 1 "${host_bits}"); do
+        pow=$((pow * 2))
+      done
+
+      num_hosts=$(( pow - 2 ))
+
+      if [ "${num_vms}" -gt "${num_hosts}" ]; then
+        echo "Error: Not enough IPs available in the subnet."
+        false
+      fi
+
+      while :; do
+        new_ip=$(generate_ip "${num_hosts}" "${ip_num}")
+        if echo " ${used_ips} " | grep -q " ${new_ip} "; then
+          continue
+        fi
+
+        used_ips="${used_ips} ${new_ip} "
+        break
+      done
+
+      lxc exec "${name}" -- ip addr add "${new_ip}/${range}" dev enp7s0
+    fi
+
+    if [ -n "${cluster_ceph_subnet}" ] && [ "${cluster_ceph_subnet}" != "${public_ceph_subnet}" ]; then
+      if [ "${num_ifaces}" -lt 4 ]; then
+        echo "Impossible to set a dedicated Ceph cluster network for less than 3 network interfaces."
+        false
+      fi
+
+      ip=$(echo "${cluster_ceph_subnet}" | cut -d '/' -f 1)
+      range=$(echo "${cluster_ceph_subnet}" | cut -d '/' -f 2)
+      IFS='.' read i1 i2 i3 i4 <<EOF
+$ip
+EOF
+
+      ip_num=$(( (i1 * 256 * 256 * 256) + (i2 * 256 * 256) + (i3 * 256) + i4 ))
+      host_bits=$(( 32 - range ))
+      pow=1
+      for _ in $(seq 1 "${host_bits}"); do
+        pow=$((pow * 2))
+      done
+
+      num_hosts=$(( pow - 2 ))
+
+      if [ "${num_vms}" -gt "${num_hosts}" ]; then
+        echo "Error: Not enough IPs available in the subnet."
+        false
+      fi
+
+      while :; do
+        new_ip=$(generate_ip "${num_hosts}" "${ip_num}")
+        if echo " ${used_ips} " | grep -q " ${new_ip} "; then
+          continue
+        fi
+
+        used_ips="${used_ips} ${new_ip} "
+        break
+      done
+
+      lxc exec "${name}" -- ip addr add "${new_ip}/${range}" dev enp8s0
```

Comment:
All of this is really really complex, and will affect all existing tests, when for most tests it should be fine to use the default values for MicroCeph's cluster and public networks.

Would it be possible to remove all of this and just call the following when running the test?
```bash
# in basic.sh, when running the test:
reset_systems 3 3 3

iface="enp8s0"
ceph_subnet="hardcoded_ip/hardcoded_range"
for m in micro01 micro02 micro03 ; do
  lxc exec "${m}" -- ip addr add "${ceph_subnet}" dev "${iface}"
done

export CEPH_PUBLIC_NETWORK="${ceph_subnet}"
microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
```

Reply:
fixed